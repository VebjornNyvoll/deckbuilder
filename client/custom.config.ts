export const env = {
  REACT_APP_FRONTEND_URL: 'http://localhost:5173/project2/',
  REACT_APP_BACKEND_URL: 'http://it2810-66.idi.ntnu.no:4000/Hearthstone/',
  CUSTOM_TEST_USERNAME: 'testUserABC123', //Use a test username that is not in the database
  CUSTOM_TEST_PASSWORD: 'testPasswordABC123',
};
