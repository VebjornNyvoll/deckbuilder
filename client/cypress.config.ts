import { defineConfig } from 'cypress';
import {env} from "./custom.config";

export default defineConfig({
  projectId: 'cvqn9t',
  retries: 4,
  e2e: {
    baseUrl:env.REACT_APP_FRONTEND_URL,
    experimentalStudio: false,
    async setupNodeEvents(on, config) {
      const localStorageCommands = await import('cypress-localstorage-commands/plugin');
      localStorageCommands.default(on, config); // Access the default property
      return config;
    },
  },
});
