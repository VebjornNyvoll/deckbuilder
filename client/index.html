<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <link rel="icon" type="image/svg+xml" href="/src/img/Icon.png" />
    <link id="theme-link" rel="stylesheet" href="/themes/lara-light-indigo/theme.css">
    <link id="theme-link-dark" rel="stylesheet" href="">
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>HearthStone Deckbuilder</title>
  </head>
  <body>
    <div id="root"></div>
    <script type="module" src="/src/main.tsx"></script>
  </body>
</html>
