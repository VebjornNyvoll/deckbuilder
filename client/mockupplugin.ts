export function viteMockPlugin() {
    return {
      name: 'mock-style-plugin',
      transform(code, id) {
        if (id.endsWith('.css') || id.endsWith('.scss')) {
          return '';
        }
      },
    };
  }
  