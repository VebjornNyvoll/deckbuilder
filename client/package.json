{
  "name": "deckbuilder",
  "private": true,
  "version": "0.0.0",
  "type": "module",
  "scripts": {
    "test": "vitest",
    "coverage": "vitest run --coverage",
    "dev": "vite",
    "prod": "vite build",
    "build": "tsc && vite build",
    "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
    "preview": "vite preview"
  },
  "dependencies": {
    "@apollo/client": "^3.8.6",
    "@reduxjs/toolkit": "^1.9.7",
    "@types/lodash.debounce": "^4.0.9",
    "@types/react-redux": "^7.1.30",
    "graphql": "^16.8.1",
    "html-react-parser": "^5.0.3",
    "jwt-decode": "^4.0.0",
    "lodash.debounce": "^4.0.8",
    "primeflex": "^3.3.1",
    "primeicons": "^6.0.1",
    "primereact": "^10.0.2",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-redux": "^8.1.3",
    "react-remove-scroll": "^2.5.7",
    "react-router-dom": "^6.16.0",
    "react-transition-group": "^4.4.5",
    "redux-mock-store": "^1.5.4"
  },
  "devDependencies": {
    "@testing-library/jest-dom": "^6.1.4",
    "@testing-library/react": "^14.1.2",
    "@testing-library/user-event": "^14.5.1",
    "@types/react": "^18.2.15",
    "@types/react-dom": "^18.2.7",
    "@types/redux-mock-store": "^1.0.6",
    "@typescript-eslint/eslint-plugin": "^6.0.0",
    "@typescript-eslint/parser": "^6.0.0",
    "@vitejs/plugin-react-swc": "^3.3.2",
    "cypress": "^13.5.1",
    "cypress-localstorage-commands": "^2.2.5",
    "eslint": "^8.45.0",
    "eslint-plugin-react-hooks": "^4.6.0",
    "eslint-plugin-react-refresh": "^0.4.3",
    "happy-dom": "^12.10.3",
    "jsdom": "^22.1.0",
    "prettier": "^3.1.0",
    "typescript": "^5.0.2",
    "vite": "^4.4.5",
    "vitest": "^0.34.6"
  }
}
