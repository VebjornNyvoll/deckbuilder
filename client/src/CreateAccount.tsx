import { Button } from "primereact/button";
import { Checkbox } from "primereact/checkbox";
import { InputText } from "primereact/inputtext";
import { useRef, useState } from "react";
import Navbar from "./components/Navbar";
import { Avatar } from "primereact/avatar";
import placeholder_avatar from './img/placeholder_avatar.png'
import { Toast } from "primereact/toast";
import { Link } from "react-router-dom";


export default function Login(){
const [checked, setChecked] = useState(false);
const [email, setEmail] = useState<string>('');
const [password, setPassword] = useState<string>('');
const [username, setUsername] = useState<string>('');

const toast = useRef<Toast>(null);

    const show = () => {
        toast.current?.show({ severity: 'info', summary: 'Terms of Service', detail: 'Man skal ikke plage andre, man skal være grei og snill, og for øvrig kan man gjøre hva man vil.' });
    };

return(
<>
<Navbar/>

<div className="flex align-items-center justify-content-center">
    <div className="surface-card p-4 shadow-2 border-round w-full lg:w-6">
        <div className="text-center mb-5">
            <Avatar image={placeholder_avatar} size="xlarge" shape="circle"/>
            <div className="text-900 text-3xl font-medium mb-3">Welcome to Deckbuilder!</div>
            <span className="text-600 font-medium line-height-3">Already have an account?</span>
            <Link to="../login" className="font-medium no-underline ml-2 text-blue-500 cursor-pointer">Log in here!</Link>
        </div>
        <form>
            <div className="card flex justify-content-center w-full mb-2">
                <span className="p-float-label w-full mb-3">
                    <InputText aria-labelledby="emailLabel" required={true} type="email" id="email" className="w-full mb-3" value={email} onChange={(e: React.ChangeEvent<HTMLInputElement>) => setEmail(e.target.value)} />
                    <label id="emailLabel" htmlFor="email">Email</label>
                </span>
            </div>
            <div className="card flex justify-content-center w-full mb-2">
                <span className="p-float-label w-full mb-3">
                    <InputText id="usernameLabel" required={true} type="password" id="username" className="w-full mb-3" value={username} onChange={(e: React.ChangeEvent<HTMLInputElement>) => setUsername(e.target.value)} />
                    <label id="usernameLabel" htmlFor="username">Username</label>
                </span>
            </div>
            <div className="card flex justify-content-center w-full mb-2">
                <span className="p-float-label w-full mb-3 card flex">
                    <InputText aria-labelledby="passwordLabel" required={true} type="password" id="password" className="w-full mb-3" value={password} onChange={(e: React.ChangeEvent<HTMLInputElement>) => setPassword(e.target.value)} />
                    <label id="usernameLabel" htmlFor="password">Password</label>
                </span>
            </div>
            <div className="flex align-items-center justify-content-between mb-6">
                <div className="flex align-items-center">
                    <Checkbox aria-labelledby="rememberMeLabel" required={true} type="checkbox" id="rememberMe" onChange={e => setChecked(e.checked)} checked={checked} className="mr-2" />
                    <Toast ref={toast} />
                    <label id="rememberMeLabel" htmlFor="rememberMe">I accept the <a href="javascript:undefined;" onClick={show}>Terms of Service</a></label>
                </div>
            </div>

            <Button label="Create account" icon="pi pi-user" className="w-full" />
        </form>
    </div>
</div>
</>
)
}
    