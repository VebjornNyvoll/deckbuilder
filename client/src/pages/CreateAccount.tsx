import { Button } from "primereact/button";
import { Checkbox } from "primereact/checkbox";
import { InputText } from "primereact/inputtext";
import { useContext, useRef, useState } from "react";
import { Avatar } from "primereact/avatar";
import placeholder_avatar from "../img/placeholder_avatar.png";
import { Toast } from "primereact/toast";
import { Link } from "react-router-dom";
import { AuthContext } from "../context/authContext";
import { useForm } from "../service/hooks";
import { useMutation } from "@apollo/client";
import { gql } from "graphql-tag";
import { useNavigate } from "react-router-dom";

const CREATE_USER = gql`
  mutation CreateUser($username: String!, $password: String!) {
    createUser(username: $username, password: $password) {
      user{username}
    }
  }
`;

function CreateAccount(props) {
  const context = useContext(AuthContext);
  const navigate = useNavigate();
  const [errors, setErrors] = useState([]);

  const validateForm = (event) => {
    event.preventDefault();

    const errors = [];

    if (!values.username || values.username.length < 3) {
      errors.push("Username needs to be minimum 3 characters");
    }

    if (!values.password || values.password.length < 3) {
      errors.push("Password needs to be minimum 3 characters");
    }

    if (!checked) {
      errors.push("You must accept the Terms of Service");
    }

    console.log("Validation Errors:", errors);

    if (errors.length === 0) {
      createUser();
    } else {
      console.log("Validation failed");
      // Display validation errors using the toast
      errors.forEach((error) => {
        toast.current?.show({
          severity: "error",
          summary: "Error Message",
          detail: error,
        });
      });
    }
  };

  const { onChange, values } = useForm({
    username: "",
    password: "",
  });

  const [createUser, { loading }] = useMutation(CREATE_USER, {
    update(proxy, { data: { createUser: userData } }) {
      context.login(userData);
      navigate("/");
    },
    onError({ graphQLErrors }) {
      setErrors(graphQLErrors);
    },
    variables: { username: values.username, password: values.password },
  });

  const [checked, setChecked] = useState(false);

  const toast = useRef<Toast>(null);

  const show = () => {
    toast.current?.show({
      severity: "info",
      summary: "Terms of Service",
      detail:
        "Man skal ikke plage andre, man skal være grei og snill, og for øvrig kan man gjøre hva man vil.",
    });
  };

  return (
    <>
      <Toast ref={toast} />
      <div className="flex align-items-center justify-content-center">
        <div className="surface-card p-4 shadow-2 border-round w-full lg:w-6">
          <div className="text-center mb-5">
            <Avatar image={placeholder_avatar} size="xlarge" shape="circle" />
            <div className="text-900 text-3xl font-medium mb-3">
              Welcome to Deckbuilder!
            </div>
            <span className="text-600 font-medium line-height-3">
              Already have an account?
            </span>
            <Link
              to="../login"
              className="font-medium no-underline ml-2 text-blue-500 cursor-pointer"
            >
              Log in here!
            </Link>
          </div>
          <form onSubmit={validateForm}>
            <div className="card flex justify-content-center w-full mb-2">
              <span className="p-float-label w-full mb-3">
                <InputText
                  required={true}
                  type="username"
                  name="username"
                  id="username"
                  className="w-full mb-3"
                  onChange={onChange}
                />
                <label id="usernameLabel" htmlFor="username">
                  Username
                </label>
              </span>
            </div>
            <div className="card flex justify-content-center w-full mb-2">
              <span className="p-float-label w-full mb-3 card flex">
                <InputText
                  aria-labelledby="passwordLabel"
                  required={true}
                  name="password"
                  type="password"
                  id="password"
                  className="w-full mb-3"
                  onChange={onChange}
                />
                <label id="usernameLabel" htmlFor="password">
                  Password
                </label>
              </span>
            </div>
            <div className="flex align-items-center justify-content-between mb-6">
              <div className="flex align-items-center">
                <Checkbox
                  aria-labelledby="termsOfServiceLabel"
                  required={true}
                  type="checkbox"
                  name="termsOfService"
                  id="termsOfService"
                  onChange={(e) => setChecked(e.checked)}
                  checked={checked}
                  className="mr-2"
                />

                <label id="termsOfServiceLabel" htmlFor="termsOfService">
                  I accept the{" "}
                  <a href="javascript:undefined;" onClick={show}>
                    Terms of Service
                  </a>
                </label>
              </div>
            </div>
            {errors.map(function (error) {
              return (
                <>
                  {toast.current?.show({
                    severity: "error",
                    summary: "Error Message",
                    detail: error.message ? error.message : error,
                  })}
                  ;
                </>
              );
            })}
            <Button
              type="submit"
              label="Create account"
              icon="pi pi-user"
              className="w-full"
            />
          </form>
        </div>
      </div>
    </>
  );
}

export default CreateAccount;
