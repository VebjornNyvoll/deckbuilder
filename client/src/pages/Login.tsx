import { Button } from "primereact/button";
import { Checkbox } from "primereact/checkbox";
import { InputText } from "primereact/inputtext";
import { useContext, useRef, useState } from "react";
import { Avatar } from "primereact/avatar";
import placeholder_avatar from '../img/placeholder_avatar.png'
import { Toast } from "primereact/toast";
import { Link } from "react-router-dom";
import { AuthContext } from "../context/authContext";
import { useForm } from "../service/hooks";
import { useMutation } from '@apollo/client';
import {gql} from 'graphql-tag';
import { useNavigate } from "react-router-dom";

const LOGIN = gql`
    mutation login($username: String!, $password: String!){
        login(username: $username, password: $password){
            user {
                username
            }
            token
        }
    }
`;

function Login(props) {
    const navigate = useNavigate();
    const context = useContext(AuthContext);
    const [errors, setErrors] = useState([]);



    const { onChange, values } = useForm({
        username: '',
        password: '',
    });

    const validateForm = (event) => {
        event.preventDefault();
    
        const errors = [];
        // Ideally neither of these errors will appear due to the HTML5 required attribute, but it's good to have a backup
        if (!values.username) {
            errors.push('Username is required');
        }
    
        if (!values.password) {
            errors.push('Password is required');
        }
    
        console.log('Validation Errors:', errors);
    
        if (errors.length === 0) {
            login();
        } else {
            // Display validation errors using the toast
            errors.forEach(error => {
                toast.current?.replace({ severity: 'error', summary: 'Error Message', detail: error });
            });
        }
    };

    const [login, { loading }] = useMutation(LOGIN, {
        update(proxy, { data: {login: userData}}) {
            context.login(userData);
            navigate('/');
        },
        onError({graphQLErrors}) {
            setErrors(graphQLErrors)
        },
        variables: {username: values.username, password: values.password}
    });
    const toast = useRef<Toast>(null);
    const showForgotPassword = () => {
        toast.current?.replace({severity: 'info', summary: 'Forgot password', detail: '✨No worries, you can just make a new account!✨'});
    }
    return (
            <>
            <Toast ref={toast} data-testid="loginToast"/>
                <div className="flex align-items-center justify-content-center">
                    <div className="surface-card p-4 shadow-2 border-round w-full lg:w-6">
                        <div className="text-center mb-5">
                            <Avatar image={placeholder_avatar} size="xlarge" shape="circle"/>
                            <div className="text-900 text-3xl font-medium mb-3">Welcome Back!</div>
                            <span className="text-600 font-medium line-height-3">Don't have an account?</span>
                            <Link to="../create-account" className="font-medium no-underline ml-2 text-blue-500 cursor-pointer">Create today!</Link>
                        </div>
                        <form onSubmit={validateForm}>
                        <div className="card flex justify-content-center w-full mb-2">
                                        <span className="p-float-label w-full mb-3">
                                            <InputText required={true} type="username" name="username" id="username" className="w-full mb-3" onChange={onChange} data-testid="loginUsername"/>
                                            <label id="usernameLabel" htmlFor="username">Username</label>
                                        </span>
                                    </div>
                                    <div className="card flex justify-content-center w-full mb-2">
                                        <span className="p-float-label w-full mb-3 card flex">
                                            <InputText aria-labelledby="passwordLabel" required={true} name="password" type="password" id="password" className="w-full mb-3" onChange={onChange} data-testid="loginPassword"/>
                                            <label id="usernameLabel" htmlFor="password">Password</label>
                                        </span>
                                    </div>
                            <div className="flex align-items-center justify-content-between mb-6">
                                <a href="javascript:undefined;" onClick={showForgotPassword} className="font-medium no-underline text-blue-500 cursor-pointer" data-testid="loginForgotPassword">Forgot your password?</a>
                            </div>
                            {errors.map(function(error){
                            return (
                                <>  
                                    {setErrors([...new Set(errors)])};
                                    {toast.current?.replace({ severity: 'error', summary: 'Error Message', detail: error.message })};
                                    {setErrors([])};
                                </>
                            )
                            })}
                            <Button type="submit" label="Sign In" icon="pi pi-user" className="w-full" data-testid="loginSubmit"/>
                        </form>
                    </div>
                </div>
            </>
    )
}



export default Login;
    