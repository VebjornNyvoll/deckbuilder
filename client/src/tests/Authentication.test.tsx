import { describe, test, expect } from 'vitest';
import { render, fireEvent, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import { ApolloProvider } from '@apollo/client';
import client from '../service/apolloClient';
import CreateAccount from '../pages/CreateAccount';
import Login from '../pages/Login';
import jsdom from 'jsdom';

const virtualConsole = new jsdom.VirtualConsole();
virtualConsole.on('error', () => {
  // No-op to skip console errors.
});

const initialState = {
  filters: {},
  sort: {},
  layout: {
    layout: 'grid',
  },
  datasaver: {
    datasaver: false,
  },

  cards: {
    cards: [],
  },
  deck: {
    triggerShowDeckEvent: false,
  },
};
const mockStore = configureStore();
const store = mockStore(initialState);

describe('Authentication behaviours', async () => {
  test('Test create user behaviour', async () => {
    render(
      <Provider store={store}>
        <ApolloProvider client={client}>
          <Router>
            <CreateAccount></CreateAccount>
          </Router>
        </ApolloProvider>
      </Provider>,
    );

    const createAccountUsername = screen.getByTestId('createAccountUsername');
    const createAccountPassword = screen.getByTestId('createAccountPassword');
    const createAccountTOS = screen.getByTestId('createAccountTOS');
    const createAccountSubmit = screen.getByTestId('createAccountSubmit');
    const createAccountTOSA = screen.getByTestId('createAccountTOSA');
    const createAccountToast = screen.getByTestId('createAccountToast');
    fireEvent.click(createAccountTOSA);
    expect(
      createAccountToast.innerHTML.match(
        'Man skal ikke plage andre, man skal være grei og snill, og for øvrig kan man gjøre hva man vil.',
      ),
    ).toBeTruthy();
    fireEvent.change(createAccountUsername, { target: { value: 'username' } });
    fireEvent.change(createAccountPassword, { target: { value: 'password' } });
    expect(createAccountTOS.checked).toBeFalsy();
    fireEvent.change(createAccountTOS, { target: { checked: true } });
    expect(createAccountTOS.checked).toBeTruthy();
    fireEvent.click(createAccountSubmit);
    expect(createAccountUsername.value).toEqual('username');
    expect(createAccountPassword.value).toEqual('password');
  });

  test('Test create user behaviour', async () => {
    render(
      <Provider store={store}>
        <ApolloProvider client={client}>
          <Router>
            <Login></Login>
          </Router>
        </ApolloProvider>
      </Provider>,
    );

    const username = screen.getByTestId('loginUsername');
    const password = screen.getByTestId('loginPassword');
    const forgotPassword = screen.getByTestId('loginForgotPassword');
    const loginSubmit = screen.getByTestId('loginSubmit');
    const loginToast = screen.getByTestId('loginToast');

    fireEvent.click(forgotPassword);
    expect(loginToast.innerHTML.match('✨No worries, you can just make a new account!✨')).toBeTruthy();

    fireEvent.change(username, { target: { value: 'username' } });
    fireEvent.change(password, { target: { value: 'password' } });

    expect(username.value).toEqual('username');
    expect(password.value).toEqual('password');

    fireEvent.click(loginSubmit);
    console.log(loginToast.innerHTML);
  });
});

describe('Create accoutn page matches snapshot', async () => {
  test('snapshot test', async () => {
    const { asFragment } = render(
      <Provider store={store}>
        <ApolloProvider client={client}>
          <Router>
            <CreateAccount></CreateAccount>
          </Router>
        </ApolloProvider>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
