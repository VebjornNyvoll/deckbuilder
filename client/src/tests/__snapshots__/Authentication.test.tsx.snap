// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Create accoutn page matches snapshot > snapshot test 1`] = `
<DocumentFragment>
  <div
    class="p-toast p-component p-toast-top-right p-ripple-disabled"
    data-pc-name="toast"
    data-pc-section="root"
    data-testid="createAccountToast"
    style="position: fixed; top: 20px; right: 20px;"
  >
    <div />
  </div>
  <div
    class="flex align-items-center justify-content-center"
  >
    <div
      class="surface-card p-4 shadow-2 border-round w-full lg:w-6"
    >
      <div
        class="text-center mb-5"
      >
        <div
          class="p-avatar p-component p-avatar-image p-avatar-circle p-avatar-xl"
          data-pc-name="avatar"
          data-pc-section="root"
        >
          <img
            alt="avatar"
            data-pc-section="image"
            src="/project2/src/img/placeholder_avatar.png"
          />
        </div>
        <div
          class="text-900 text-3xl font-medium mb-3"
        >
          Welcome to Deckbuilder!
        </div>
        <span
          class="text-600 font-medium line-height-3"
        >
          Already have an account?
        </span>
        <a
          class="font-medium no-underline ml-2 text-blue-500 cursor-pointer"
          href="/login"
        >
          Log in here!
        </a>
      </div>
      <form>
        <div
          class="card flex justify-content-center w-full mb-2"
        >
          <span
            class="p-float-label w-full mb-3"
          >
            <input
              class="p-inputtext p-component w-full mb-3 w-full mb-3"
              data-pc-name="inputtext"
              data-pc-section="root"
              data-testid="createAccountUsername"
              id="username"
              name="username"
              required=""
              type="username"
            />
            <label
              for="username"
              id="usernameLabel"
            >
              Username
            </label>
          </span>
        </div>
        <div
          class="card flex justify-content-center w-full mb-2"
        >
          <span
            class="p-float-label w-full mb-3 card flex"
          >
            <input
              aria-labelledby="passwordLabel"
              class="p-inputtext p-component w-full mb-3 w-full mb-3"
              data-pc-name="inputtext"
              data-pc-section="root"
              data-testid="createAccountPassword"
              id="password"
              name="password"
              required=""
              type="password"
            />
            <label
              for="password"
              id="usernameLabel"
            >
              Password
            </label>
          </span>
        </div>
        <div
          class="flex align-items-center justify-content-between mb-6"
        >
          <div
            class="flex align-items-center"
          >
            <div
              class="mr-2 p-checkbox p-component"
              data-pc-name="checkbox"
              data-pc-section="root"
              data-testid="createAccountTOS"
              id="termsOfService"
              type="checkbox"
            >
              <div
                class="p-hidden-accessible"
                data-pc-section="hiddeninputwrapper"
              >
                <input
                  aria-labelledby="termsOfServiceLabel"
                  data-pc-section="hiddeninput"
                  name="termsOfService"
                  required=""
                  type="checkbox"
                />
              </div>
              <div
                class="p-checkbox-box"
                data-p-disabled="false"
                data-p-focus="false"
                data-p-highlight="false"
                data-pc-section="input"
              />
            </div>
            <label
              for="termsOfService"
              id="termsOfServiceLabel"
            >
              I accept the 
              <a
                data-testid="createAccountTOSA"
                href="javascript:undefined;"
              >
                Terms of Service
              </a>
            </label>
          </div>
        </div>
        <button
          aria-label="Create account"
          class="w-full p-button p-component"
          data-pc-name="button"
          data-pc-section="root"
          data-testid="createAccountSubmit"
          type="submit"
        >
          <span
            class="p-button-icon p-c p-button-icon-left pi pi-user"
            data-pc-section="icon"
          />
          <span
            class="p-button-label p-c"
            data-pc-section="label"
          >
            Create account
          </span>
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
